pre_commit/commands/run.py

```diff
@@ -171,12 +172,16 @@ def _run_single_hook(
             filenames = ()
         time_before = time.time()
         language = languages[hook.language]
-        retcode, out = language.run_hook(hook, filenames, use_color)
-        duration = round(time.time() - time_before, 2) or 0
-        diff_after = _get_diff()
+        with monitor.trace(f'precommit.hook.{hook.name}') as trace:
+            retcode, out = language.run_hook(hook, filenames, use_color)
+
+            duration = round(time.time() - time_before, 2) or 0
+            diff_after = _get_diff()
+
+            # if the hook makes changes, fail the commit
+            files_modified = diff_before != diff_after
+            trace.set_success(not (retcode or files_modified))
```

Comment:
nit: might be worth pulling `retcode or files_modified` into a variable? But I also don't know how this changes with the "allow changes to be committed" change.

Reply:
Definitely would be, but being lazy here because that's done in the other change (and it's not that terrible in the brief interim).